### DISCUSSION TITLE
How do I get keyboard input working on Wayland?

### DISCUSSION THREAD

**Discussion by Hubro** [rustdesk/rustdesk#2928] [Q&A] [answered]
The readme has some sparse instructions on keyboard input on Wayland, namely running `sudo rustdesk --service` before starting rustdesk.

I've done this, but keyboard input still doesn't work. Mouse input works as expected, though.

```
➜ ps ax | grep -i rustdesk
1699577 pts/3    S+     0:00 sudo rustdesk --service
1699578 pts/3    Sl+    0:02 rustdesk --service
1701630 pts/4    Rl     1:58 /usr/bin/rustdesk --connect 226494975
1702018 pts/3    Sl+    0:01 /usr/bin/rustdesk --server
1714351 pts/2    S+     0:00 grep -i rustdesk
```

To clarify, I'm on Wayland (Sway) and I'm connecting to a Gnome (X11) Linux desktop. Screen sharing and mouse control works perfectly, but the keyboard is completely ignored.

Is there anything obvious I'm doing wrong? What can I do to troubleshoot further?

---

Rustdesk 1.1.9
Arch Linux (6.1.3-arch1-1)

**Comment by rustdesk** [comment, marked as answer]
try out nightly build

**Reply by Hubro** [reply]
Oh hey, neat! It works in 1.2.0 :smile: Installed using `rustdesk-nightly` from AUR
